dvc/external_repo.py

```diff
@@ -127,8 +127,12 @@ def _fix_upstream(self):
         if not os.path.isdir(self.url):
             return
 
+        try:
+            src_repo = Repo(self.url)
+        except NotDvcRepoError:
+            return
+
         remote_name = self.config["core"].get("remote")
```

Comment:
Ideally this should be under the `try&finally`, so `src_repo` get's closed :) Nitpicky but still.

Reply:
`src_repo` cannot be in finally if we keep in below `try`. So, I very much prefer this easier flow control.